scripts/train.py

```diff
@@ -31,30 +42,53 @@
         "save_every": args.save_every
     })
 
-    loaders = fetch_loaders(data_dir, args.batch_size)
+    loaders = fetch_loaders(data_dir, args.batch_size, shuffle=True)
+
+    # TODO:handle this error better
+    # if input mask dimension different than outchannels
+    outchannels = conf.model_opts.args.outchannels
+    y_channels = [y.shape[-1] for _, y in loaders["val"]][0]
+    if y_channels != outchannels:
+        raise ValueError("Output dimension is different from model outchannels.")
+
+    # TODO: try to have less nested if/else
```

Comment:
I think it would be clearer to just hard code loss_fn to the value on line 60 here. This is just a script, not a package function, so it doesn't need to be entirely general. If people want to try many different loss functions, we could create a package helper to fetch specific cases, but until that need arises, I don't think we need to write code to handle it.

Reply:
I agree with that we don't need much generality in this script. This helps, for now, with the error if someone uses different channels than specified in the config file. This error will run and won't be caught due to some size manipulation in the rest of the library. It also has occurred with us, so I assume it can occur to another user easily.  
We can move it to a more suitable place though. It was an ad-hoc solution to a problem that happen to multiple people.

Reply:
I see, if the check is needed for current code to work, there's no need to refactor it now. You can merge it whenever you would like.